client/modules/IDE/components/FileNode.jsx

```diff
@@ -103,6 +118,7 @@ export class FileNode extends React.Component {
 
   hideEditFileName() {
     this.setState({ isEditingName: false });
+    this.commitFileNameChange();
```

Comment:
Because this is called `onBlur`, along with `validateFileName`, adding `this.commitFileNameChange()` here is actually causing `this.props.updateFilename` to be called multiple times. i think this line and function is unnecessary!

Reply:
This call is actually really important, It's what triggers change detection!
I noticed `validateFileName` only calls `this.props.updateFilename` when the filename is problematic, to reset it to the original value.
Keeping this line as it is would skip filename validation entirely, which is very problematic, so I moved it into `validateFileName`. Now it's only called when the filename is valid, and no duplicate calls occur :)